### ISSUE TITLE
NgbModalModule + NgbButtonsModule = ExpressionChangedAfterItHasBeenCheckedError by default

### ISSUE THREAD

**Issue by bcmedeiros** [ng-bootstrap/ng-bootstrap#2870]
### Bug description:

Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'focus: undefined'. Current value: 'focus: true'.

It seems that https://github.com/ng-bootstrap/ng-bootstrap/commit/10fd5e4 introduced a new bug for calling `elementToFocus.focus()` during `ngAfterViewInit()` lifecycle phase.

Can you check that, please?

### Link to minimally-working StackBlitz that reproduces the issue:
https://stackblitz.com/edit/angular-2cvt6e-igmht4

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 7.0.4

ng-bootstrap: 4.0.0

Bootstrap: 4.1.3

**Comment by brunojcm** [comment]
@pkozlowski-opensource, thanks for the quick response!

But I have to confess, I'm a little disappointed with it. I'll try to explain why. 

I understand that the root cause of the issue is the same for this and #1252, BUT... 

1. Closing it as a duplicate is a little bit unfair. Everything was working fine here in my application before I upgraded to angular 7, which requires ng-bootstrap 4.0.0. Ng-bootstrap team deliberately chose not add the `focusFirst` global configuration, included in the original design of the focus feature, and now I have to either find a solution myself for each individual modal I have (or accept the error happening every time) or revert all the stack upgrade I just did, including retuning to Angular 6 and ng-bootstrap 2.2.2.

2. The title for #1252 is rather specific.I kind of feel bad reporting "duplicated" issues, I have the feeling that I didn't search well before reporting, but in this case it would be very hard to relate both issues. I think that, if #1252 is going to be the centralised place to discuss the matter, it deserves a more appropriate and generic name, as its scope is getting broader.
